src/axelrod_fortran/player.py

```diff
@@ -1,20 +1,114 @@
+from collections import defaultdict
+from ctypes import cdll, c_int, c_float, byref, POINTER
+from ctypes.util import find_library
+import os
 import random
+import shutil
+import tempfile
+import uuid
 import warnings
 
 import axelrod as axl
 from axelrod.interaction_utils import compute_final_score
 from axelrod.action import Action
-from ctypes import cdll, c_int, c_float, byref, POINTER
 from .strategies import characteristics
 
 C, D = Action.C, Action.D
 actions = {0: C, 1: D}
 original_actions = {C: 0, D: 1}
 
 
+self_interaction_message = """
+You are playing a match with the same player against itself. However
+axelrod_fortran players share memory. You can initialise another instance of an
+Axelrod_fortran player with player.clone().
+"""
+
+
+class LibraryManager(object):
+    """LibraryManager creates and loads copies of a shared library, which
+    enables multiple copies of the same strategy to be run without the end user
+    having to maintain many copies of the shared library.
+
+    This works by making a copy of the shared library file and loading it into
+    memory again. Loading the same file again will return a reference to the
+    same memory addresses.
+
+    Additionally, library manager tracks how many copies of the library have
+    been loaded, and how many copies there are of each Player, so as to load
+    only as many copies of the shared library as needed.
+    """
+
+    def __init__(self, shared_library_name, verbose=False):
+        self.shared_library_name = shared_library_name
+        self.verbose = verbose
+        self.library_copies = []
+        self.player_indices = defaultdict(set)
+        self.player_next = defaultdict(set)
+        # Generate a random prefix for tempfile generation
+        self.prefix = str(uuid.uuid4())
```

Comment:
I can't remembery why, but I seem to recall it's safer to use uuid4().hex here instead of str

Reply:
Based on the docs [here](https://docs.python.org/3/library/uuid.html) it looks like `str(...)` gives the hex value (perhaps previously it was different?). I just need a unique value so we can change to `hex` if there are any concerns.